compass/landice/tests/ensemble_generator/plot_ensemble.py

```diff
@@ -202,61 +310,149 @@
 
     fpath = run + "/output/globalStats.nc"
     if os.path.exists(fpath):
+        n_with_output += 1
         f = netCDF4.Dataset(fpath, 'r')
         years = f.variables['daysSinceStart'][:] / 365.0
 
-        VAF = f.variables['volumeAboveFloatation'][:]
-        SLR = (VAF[0] - VAF) / 3.62e14 * rhoi / rhosw * 1000.
+        final_time[idx] = years[-1]
+
+        VAF = f.variables['volumeAboveFloatation'][:] / \
+            (1.0e12 / rhoi)  # in Gt
         totalArea = f.variables['totalIceArea'][:] / 1.0e6  # in km2
         fltArea = f.variables['floatingIceArea'][:] / 1.0e6  # in km2
         grdArea = f.variables['groundedIceArea'][:] / 1.0e6  # in km2
-        groundingLineFlux = f.variables['groundingLineFlux'][:] / 1.0e12  # Gt
-        BMB = f.variables['totalFloatingBasalMassBal'][:] / -1.0e12  # in Gt
-
-        # find target year index
-        indices = np.nonzero(years >= targetYear)[0]
-
-        # color lines depending on if they match obs or not
-        col = 'k'
-        alph = 0.2
-        GLobs = qoi_info['GL flux']['obs']
-        if GLobs is not None and len(indices) > 0:
+        iceArea = f.variables['totalIceArea'][:] / 1000.0**2  # in km^2
+        grdVol = f.variables['groundedIceVolume'][:] / (1.0e12 / rhoi)  # in Gt
+        BMB = f.variables['totalFloatingBasalMassBal'][:] / -1.0e12  # in Gt/yr
+        grSMB = f.variables['totalGroundedSfcMassBal'][:] / -1.0e12  # in Gt/yr
+        groundingLineFlux = f.variables['groundingLineFlux'][:] \
+            / 1.0e12  # Gt/yr
+        GLMigFlux = f.variables['groundingLineMigrationFlux'][:] \
+            / 1.0e12  # Gt/yr
+        w = 50
+        GLMigFlux2 = np.convolve(GLMigFlux, np.ones(w), 'same') / w
+        GLflux2 = groundingLineFlux + GLMigFlux2
```

Comment:
This line gives an error when I run it: `ValueError: operands could not be broadcast together with shapes (42,) (50,)`. It seems like the hardcording of `w=50` in line 332 is causing issues because `groundingLineFlux` has different length in this instance

Reply:
Yeah - I think this just happens when the the number of time levels in a run is less than `w`.  This smoothing operation was a bit of hack to get decent looking GL flux time series plotted.  It could use further improvement, but I'd rather not dig into it in this PR.

We could have `w` shorten to the length of the time-series if needed, but then different runs might have different levels of smoothing applied.  I've more comments about this issue in the script, so if this error occurs and a user looks at the script where the error occurred, the comments should guide them how to resolve it.